packages/webapp/src/containers/Task/saga.js

```diff
@@ -609,11 +613,16 @@ export function* createTaskSaga({ payload }) {
         task_translation_key === 'HARVEST_TASK' ? result.data[0] : result.data;
       yield call(getTasksSuccessSaga, { payload: isHarvest ? result.data : [result.data] });
       if (alreadyCompleted) {
+        const isCustomTaskWithAnimals =
+          isCustomTask && (result.data.animals?.length || result.data.animal_batches?.length);
         yield call(onReqSuccessSaga, {
           message: i18n.t('message:TASK.CREATE.SUCCESS'),
-          pathname: getTaskCompletePathname(task_id, task_translation_key),
+          pathname: getTaskCompletePathname(task_id, task_translation_key, isCustomTaskWithAnimals),
         });
         if (isCustomTask) {
+          yield put(
+            setPersistedPaths([`/tasks/${task_id}/complete`, `/tasks/${task_id}/before_complete`]),
+          );
           yield put(setFormData({ task_id, taskType }));
```

Comment:
Are you able to create and complete a custom task without animals? The solution looks solid, but `persistedFormData` is still not updating for me, and I can't complete the task.
I tried adding `yield delay(500);` before `setFormData`, and that seemed to work.
My laptop feels very slow today, so the issue might not be reproducible on your end.

Reply:
@SayakaOno yeah, I'm able to complete it without a delay (the form data does populate). I hate that it's system dependent, that sucks 😦

> My laptop feels very slow today

Unrelated but did you update to Sequoia recently? 😅

Reply:
`500` was just a random number, but I guess it's reasonable?

I think it's been a few weeks since I updated to Sequoia. I didn't do anything, but my laptop is running much faster now, and it works fine without a delay 😅